HLTrigger/special/plugins/HLTPixelThrustFilter.cc

```diff
@@ -0,0 +1,107 @@
+#include "HLTrigger/HLTcore/interface/HLTFilter.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "DataFormats/HLTReco/interface/TriggerFilterObjectWithRefs.h"
+#include "DataFormats/SiPixelCluster/interface/SiPixelCluster.h"
+#include "RecoLocalTracker/ClusterParameterEstimator/interface/PixelClusterParameterEstimator.h"
+#include "Geometry/TrackerGeometryBuilder/interface/TrackerGeometry.h"
+#include "Geometry/Records/interface/TrackerDigiGeometryRecord.h"
+#include "RecoLocalTracker/Records/interface/TkPixelCPERecord.h"
+#include "PhysicsTools/CandUtils/interface/Thrust.h"
+
+//
+// class declaration
+//
+
+class HLTPixelThrustFilter : public HLTFilter {
+public:
+  explicit HLTPixelThrustFilter(const edm::ParameterSet&);
+  ~HLTPixelThrustFilter() override;
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  bool hltFilter(edm::Event&,
+                 const edm::EventSetup&,
+                 trigger::TriggerFilterObjectWithRefs& filterproduct) const override;
+  edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> const trackerGeometryRcdToken_;
+  edm::ESGetToken<PixelClusterParameterEstimator, TkPixelCPERecord> const pixelCPERcdToken_;
+  edm::InputTag inputTag_;  // input tag identifying product containing pixel clusters
+  double min_thrust_;       // minimum thrust
+  double max_thrust_;       // maximum thrust
```

Comment:
```suggestion
  const edm::ESGetToken<TrackerGeometry, TrackerDigiGeometryRecord> const trackerGeometryRcdToken_;
  const edm::ESGetToken<PixelClusterParameterEstimator, TkPixelCPERecord> const pixelCPERcdToken_;
  const edm::InputTag inputTag_;  // input tag identifying product containing pixel clusters
  const double min_thrust_;       // minimum thrust
  const double max_thrust_;       // maximum thrust
```

Reply:
Implemented in 31fb30855064b1366f53d5483f7d7402e7c6ddc1